# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: toolkit/components/credentialmanagement/IdentityCredentialStorageService.h
# Commit: 459739221a88
# Full Hash: 459739221a884b2b1c2adab42fd19fafe279bd20
# Author: Sandor Molnar <smolnar@mozilla.com>
# Date: 2025-06-05 04:07:37
# Regressor Bug: 1969017
# File Overlap Count: 2
# Description:
#   Revert "Bug 1969015, 1969017: apply code formatting via Lando" for causing build bustages
#   
#   This reverts commit ab681e36aeafc21ba62bb4680bdaa6206f02546d.
#   
#   Revert "Bug 1969017 - Refactor FedCM to use a managed actor rather than tagging onto WindowGlobalParent - r=anti-tracking-reviewers,emz"
# ==============================================================================

diff -r fc9986230203 -r 459739221a88 toolkit/components/credentialmanagement/IdentityCredentialStorageService.h
--- a/toolkit/components/credentialmanagement/IdentityCredentialStorageService.h	Thu Jun 05 00:15:19 2025 +0300
+++ b/toolkit/components/credentialmanagement/IdentityCredentialStorageService.h	Thu Jun 05 00:19:41 2025 +0300
@@ -14,6 +14,7 @@
 #include "mozilla/OriginAttributes.h"
 #include "mozIStorageConnection.h"
 #include "mozIStorageFunction.h"
+#include "mozilla/dom/IPCIdentityCredential.h"
 #include "nsIAsyncShutdown.h"
 #include "nsIFile.h"
 #include "nsIIdentityCredentialStorageService.h"
@@ -86,6 +87,8 @@
   // Grab all data from the disk database and insert it into the memory
   // database/ This is used at start up
   nsresult LoadMemoryTableFromDisk();
+  nsresult LoadLightweightMemoryTableFromDisk();
+  nsresult LoadHeavyweightMemoryTableFromDisk();
 
   // Used to (thread-safely) track how many operations have been launched to the
   // worker thread so that we can wait for it to hit zero before close the disk
@@ -119,6 +122,14 @@
 
   static nsresult ClearData(mozIStorageConnection* aDatabaseConnection);
 
+  static nsresult UpsertLightweightData(
+      mozIStorageConnection* aDatabaseConnection,
+      const dom::IPCIdentityCredential& aData);
+
+  static nsresult DeleteLightweightData(
+      mozIStorageConnection* aDatabaseConnection,
+      const dom::IPCIdentityCredential& aData);
+
   static nsresult DeleteDataFromOriginAttributesPattern(
       mozIStorageConnection* aDatabaseConnection,
       OriginAttributesPattern const& aOriginAttributesPattern);